cli/flags.rs

```diff
@@ -2654,12 +2657,20 @@ fn test_parse(flags: &mut Flags, matches: &clap::ArgMatches) {
   }
 
   let concurrent_jobs = if matches.is_present("jobs") {
+    println!(
+      "{}",
+      crate::colors::yellow("Warning: --jobs flag is deprecated. Use 'DENO_JOBS' environment variable instead."),
+    );
     if let Some(value) = matches.value_of("jobs") {
       value.parse().unwrap()
     } else {
       std::thread::available_parallelism()
         .unwrap_or(NonZeroUsize::new(1).unwrap())
     }
+  } else if let Ok(value) = env::var("DENO_JOBS") {
+    value
+      .parse::<NonZeroUsize>()
```

Comment:
Here we just ignore error when parsing value fails. Instead we can print error message asking to use valid value. What do you think? @dsherret @nayeemrmn

Reply:
Makes sense to me. Probably not worth any extra complexity as someone putting in a non-number value would be extremely rare.